python/run-tests.py

```diff
@@ -22,22 +22,28 @@
 from os import path
 
 
-def test(root_dir, package):
-    # Run all of the test under test/python directory, each of them
+def test(root_dir, test_path, package):
+    # Run all of the tests under test_path directory, each of them
     # has main entry point to execute, which is python's unittest testing
     # framework.
     python_root_dir = path.join(root_dir, "python")
-    test_dir = path.join(python_root_dir, path.join("delta", "tests"))
+    test_dir = path.join(python_root_dir, path.join(test_path, "tests"))
     test_files = [os.path.join(test_dir, f) for f in os.listdir(test_dir)
                   if os.path.isfile(os.path.join(test_dir, f)) and
                   f.endswith(".py") and not f.startswith("_")]
-    extra_class_path = path.join(python_root_dir, path.join("delta", "testing"))
+    extra_class_path = path.join(python_root_dir, path.join(test_path, "testing"))
 
     for test_file in test_files:
         try:
             cmd = ["spark-submit",
                    "--driver-class-path=%s" % extra_class_path,
-                   "--packages", package, test_file]
+                   "--packages", package +
+                   ",io.delta:delta-connect-server_2.13:4.0.0-SNAPSHOT," +
```

Comment:
How does this work? Does it use the locally built packages for this? Will it continue to use the local packages once we publish these packages to Maven?

Reply:
@tomvanbussel My understanding is that this downloads from `Maven`.

> WARNING: Using incubator modules: jdk.incubator.vector
> :: loading settings :: url = jar:file:/home/runner/.local/share/virtualenvs/delta-Jo9PrCI6/lib/python3.9/site-packages/pyspark/jars/ivy-2.5.2.jar!/org/apache/ivy/core/settings/ivysettings.xml
> Ivy Default Cache set to: /home/runner/.ivy2.5.2/cache
> The jars for the packages stored in: /home/runner/.ivy2.5.2/jars
> io.delta#delta-spark_2.13 added as a dependency
> io.delta#delta-connect-server_2.13 added as a dependency
> io.delta#delta-connect-common_2.13 added as a dependency
> org.apache.spark#spark-connect_2.13 added as a dependency
> org.apache.spark#spark-connect-client-jvm_2.13 added as a dependency
> org.apache.spark#spark-connect-common_2.13 added as a dependency
> com.google.protobuf#protobuf-java added as a dependency

> downloading file:/home/runner/.m2/repository/io/delta/delta-connect-server_2.13/4.0.0-SNAPSHOT/delta-connect-server_2.13-4.0.0-SNAPSHOT.jar ...

> 	[SUCCESSFUL ] io.delta#delta-connect-server_2.13;4.0.0-SNAPSHOT!delta-connect-server_2.13.jar (1ms)
> downloading file:/home/runner/.m2/repository/io/delta/delta-connect-common_2.13/4.0.0-SNAPSHOT/delta-connect-common_2.13-4.0.0-SNAPSHOT.jar ...
> 	[SUCCESSFUL ] io.delta#delta-connect-common_2.13;4.0.0-SNAPSHOT!delta-connect-common_2.13.jar (3ms)
> downloading https://repo1.maven.org/maven2/org/apache/spark/spark-connect_2.13/4.0.0-preview1/spark-connect_2.13-4.0.0-preview1.jar ...
> 	[SUCCESSFUL ] org.apache.spark#spark-connect_2.13;4.0.0-preview1!spark-connect_2.13.jar (94ms)
> downloading https://repo1.maven.org/maven2/org/apache/spark/spark-connect-client-jvm_2.13/4.0.0-preview1/spark-connect-client-jvm_2.13-4.0.0-preview1.jar ...
> 	[SUCCESSFUL ] org.apache.spark#spark-connect-client-jvm_2.13;4.0.0-preview1!spark-connect-client-jvm_2.13.jar (144ms)
> downloading https://repo1.maven.org/maven2/org/apache/spark/spark-connect-common_2.13/4.0.0-preview1/spark-connect-common_2.13-4.0.0-preview1.jar ...
> 	[SUCCESSFUL ] org.apache.spark#spark-connect-common_2.13;4.0.0-preview1!spark-connect-common_2.13.jar (37ms)
> downloading https://repo1.maven.org/maven2/com/google/protobuf/protobuf-java/3.25.1/protobuf-java-3.25.1.jar ...
> 	[SUCCESSFUL ] com.google.protobuf#protobuf-java;3.25.1!protobuf-java.jar (19ms)
> downloading https://repo1.maven.org/maven2/io/grpc/grpc-protobuf/1.62.2/grpc-protobuf-1.62.2.jar ...
> 	[SUCCESSFUL ] io.grpc#grpc-protobuf;1.62.2!grpc-protobuf.jar (4ms)

Reply:
Okay, but that's not really what we want. We want these tests to use the locally built package when available. Otherwise you'd have to publish a new Maven artifact every time the server plugin is modified.

Reply:
@tomvanbussel Added as a TODO in the future that should be done after the Preview.